build.go

```diff
@@ -191,11 +186,36 @@ func Build(builder Builder, options ...Option) {
 		config.exitHandler.Error(fmt.Errorf("this version of libcnb is only compatible with buildpack APIs >= %s, <= %s", MinSupportedBPVersion, MaxSupportedBPVersion))
 		return
 	}
+	var buildpackPlanPath string
 
-	ctx.Layers = Layers{config.arguments[1]}
-	logger.Debugf("Layers: %+v", ctx.Layers)
+	if API.LessThan(semver.MustParse("0.8")) {
+		if len(config.arguments) != 4 {
+			config.exitHandler.Error(fmt.Errorf("expected 3 arguments and received %d", len(config.arguments)-1))
+			return
+		}
+		ctx.Layers = Layers{config.arguments[1]}
+		ctx.Platform.Path = config.arguments[2]
+		buildpackPlanPath = config.arguments[3]
+	} else {
+		layersDir, ok := os.LookupEnv("CNB_LAYERS_DIR")
+		if !ok {
+			config.exitHandler.Error(fmt.Errorf("expected CNB_LAYERS_DIR to be set"))
+			return
+		}
+		ctx.Layers = Layers{layersDir}
+		ctx.Platform.Path, ok = os.LookupEnv("CNB_PLATFORM_DIR")
+		if !ok {
+			config.exitHandler.Error(fmt.Errorf("expected CNB_PLATFORM_DIR to be set"))
+			return
+		}
+		buildpackPlanPath, ok = os.LookupEnv("CNB_BP_PLAN_PATH")
+		if !ok {
+			config.exitHandler.Error(fmt.Errorf("expected CNB_BP_PLAN_PATH to be set"))
+			return
+		}
+	}
 
-	ctx.Platform.Path = config.arguments[2]
+	logger.Debugf("Layers: %+v", ctx.Layers)
```

Comment:
```suggestion
	logger.Debugf("Layers Directory: %+v", ctx.Layers)
```

Reply:
Is it okay if we keep it to the previous statement since that is what libcnb was logging before?

Reply:
Oh, I thought that had changed in the diff. Yes, not a huge issue. Just thought that would be a little clearer.